### ISSUE TITLE
ONNX dynamo export: meshgrid is broken in the nightly build

### ISSUE THREAD

**Issue by Bludator** [pytorch/pytorch#141770]
### 🐛 Describe the bug

Runs in the stable version, does not run in the nightly version:
```python
import torch

class Model(torch.nn.Module):
	def __init__(self):
		super().__init__()

	def forward(self, x,y):
		out = torch.meshgrid(x, y, indexing="xy")
		return out

onnx_program = torch.onnx.export(Model(), (torch.tensor(20),torch.tensor(20)), dynamo=True)
```

~~Without the `indexing="xy"` it gives error:~~

> ~~<class 'TypeError'>: meshgrid() missing 1 required keyword-only argument: 'indexing'~~

I just realize there is also this warning so it could be expected:

> .venv_torch_nightly/lib/python3.12/site-packages/torch/functional.py:539: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at /pytorch/aten/src/ATen/native/TensorShape.cpp:3611.)

However, even if I set the `indexing` parameter I am getting an error:

> <class 'RuntimeError'>: Found a custom (non-ATen) operator whose output has alias annotations: prims::broadcast_in_dim(Tensor(a) a, SymInt[] shape, int[] broadcast_dimensions) -> Tensor(a). We only support functionalizing operators whose outputs do not have alias annotations (e.g. 'Tensor(a)' is a Tensor with an alias annotation whereas 'Tensor' is a Tensor without. The '(a)' is the alias annotation). The alias annotation specifies that the output Tensor shares storage with an input that has the same annotation. Please check if 
> (1) the output needs to be an output (if not, don't return it), 
> (2) if the output doesn't share storage with any inputs, then delete the alias annotation. 
> (3) if the output indeed shares storage with an input, then add a .clone() before returning it to prevent storage sharing and then delete the alias annotation. Otherwise, please file an issue on GitHub.

### Versions

<details>
<summary>Versions & HW</summary>
PyTorch version: 2.6.0.dev20241128+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.1 LTS (x86_64)
GCC version: (Ubuntu 13.2.0-23ubuntu4) 13.2.0
Clang version: Could not collect
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0] (64-bit runtime)
Python platform: Linux-6.8.0-49-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1050 Ti
Nvidia driver version: 560.35.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        36 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               4
On-line CPU(s) list:                  0-3
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) i5-2400 CPU @ 3.10GHz
CPU family:                           6
Model:                                42
Thread(s) per core:                   1
Core(s) per socket:                   4
Socket(s):                            1
Stepping:                             7
CPU(s) scaling MHz:                   70%
CPU max MHz:                          3400.0000
CPU min MHz:                          1600.0000
BogoMIPS:                             6186.13
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx lahf_lm epb pti ssbd ibrs ibpb stibp tpr_shadow flexpriority ept vpid xsaveopt dtherm ida arat pln pts vnmi md_clear flush_l1d
Virtualization:                       VT-x
L1d cache:                            128 KiB (4 instances)
L1i cache:                            128 KiB (4 instances)
L2 cache:                             1 MiB (4 instances)
L3 cache:                             6 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-3
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          KVM: Mitigation: VMX disabled
Vulnerability L1tf:                   Mitigation; PTE Inversion; VMX conditional cache flushes, SMT disabled
Vulnerability Mds:                    Mitigation; Clear CPU buffers; SMT disabled
Vulnerability Meltdown:               Mitigation; PTI
Vulnerability Mmio stale data:        Unknown: No mitigations
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP disabled; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.1.2
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-cusparselt-cu12==0.6.2
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] onnx==1.17.0
[pip3] onnxruntime==1.20.1
[pip3] onnxscript==0.1.0.dev20241128
[pip3] pytorch-triton==3.2.0+git35c6c7c6
[pip3] torch==2.6.0.dev20241128+cu124
[pip3] torchaudio==2.5.0.dev20241128+cu124
[pip3] torchvision==0.20.0.dev20241128+cu124
[conda] Could not collect
</details>

**Comment by Bludator** [comment]
Well, it could be ~fixed~ hacked by switching the lines [61 and 62 deep in the onnx exporter ⛏️](https://github.com/pytorch/pytorch/blob/7224cd4471260c1cb0d44725c13bbeb65773e161/torch/onnx/_internal/exporter/_decomp.py#L61)
After lot of debugging it turns out that these two lines produces decompositions somewhat differently. The `torch.export.default_decompositions()` are wrapped in `functools.partial` the other not. As for the meshgrid, both calls gives dict with decomposition for it but the second one raise the error. The other do not but it is overridden.

I would love to get some insight to why I am getting the error, how the composition dicts are compiled etc.

**Comment by titaiwangms** [comment]
@justinchuby Maybe we can get rid of `torch._decomp.decomposition_table.items()` given that we already have `torch.export.default_decompositions().items()` which seems to be a proper api from torch.export?

**Comment by justinchuby** [comment]
@tugsbayasgalan do you have a recommendation? Thanks!

**Comment by justinchuby** [comment]
> @justinchuby Maybe we can get rid of `torch._decomp.decomposition_table.items()` given that we already have `torch.export.default_decompositions().items()` which seems to be a proper api from torch.export?

We can compare the contents and make sure it is safe to do so.

**Comment by titaiwangms** [comment]
These are the mismatched op function brtween these two table:

{<OpOverload(op='aten.std', overload='names_dim')>, <OpOverload(op='aten.margin_ranking_loss', overload='default')>, <OpOverload(op='aten.fft_rfft', overload='default')>, <OpOverload(op='aten.silu_backward', overload='default')>, <OpOverload(op='aten.var', overload='correction_names')>, <OpOverload(op='aten.any', overload='dimname')>, <OpOverload(op='aten.fft_irfftn', overload='default')>, <OpOverload(op='aten.fft_hfft2', overload='default')>, <OpOverload(op='aten.index_select', overload='dimname')>, <OpOverload(op='aten.mish_backward', overload='default')>, <OpOverload(op='aten.fft_fft', overload='default')>, <OpOverload(op='aten.norm', overload='names_ScalarOpt_dim')>, <OpOverload(op='aten.fft_rfftn', overload='default')>, <OpOverload(op='aten.fft_rfft2', overload='default')>, <OpOverload(op='aten.all', overload='dimname')>, <OpOverload(op='aten.var_mean', overload='default')>, <OpOverload(op='aten.fft_ihfft2', overload='default')>, <OpOverload(op='aten.var', overload='default')>, <OpOverload(op='aten.index_fill', overload='Dimname_Tensor')>, <OpOverload(op='aten.std', overload='dim')>, <OpOverload(op='aten.index_copy', overload='dimname')>, <OpOverload(op='aten.fft_fftn', overload='default')>, <OpOverload(op='aten.meshgrid', overload='default')>, <OpOverload(op='aten.item', overload='default')>, <OpOverload(op='aten.meshgrid', overload='indexing')>, <OpOverload(op='aten.index_fill', overload='Dimname_Scalar')>, <OpOverload(op='aten.fft_hfftn', overload='default')>, <OpOverload(op='aten.std', overload='default')>, <OpOverload(op='aten.fft_hfft', overload='default')>, <OpOverload(op='aten.fft_fftshift', overload='default')>, <OpOverload(op='aten.pairwise_distance', overload='default')>, <OpOverload(op='aten.var_mean', overload='names_dim')>, <OpOverload(op='aten.fft_irfft2', overload='default')>, <OpOverload(op='aten.fft_ifftshift', overload='default')>, <OpOverload(op='aten.fft_irfft', overload='default')>, <OpOverload(op='aten.var_mean', overload='correction_names')>, <OpOverload(op='aten.var', overload='dim')>, <OpOverload(op='aten.fft_ihfftn', overload='default')>, <OpOverload(op='aten.std_mean', overload='default')>, <OpOverload(op='aten.pdist', overload='default')>, <OpOverload(op='aten.special_ndtr', overload='default')>, <OpOverload(op='aten.fft_ifftn', overload='default')>, <OpOverload(op='aten.fft_ifft2', overload='default')>, <OpOverload(op='aten.std_mean', overload='dim')>, <OpOverload(op='aten.fft_ifft', overload='default')>, <OpOverload(op='aten.var_mean', overload='dim')>, <OpOverload(op='aten.cumprod', overload='dimname')>, <OpOverload(op='aten.fft_fft2', overload='default')>, <OpOverload(op='aten.logsumexp', overload='names')>, <OpOverload(op='aten.is_complex', overload='default')>, <OpOverload(op='aten.fft_ihfft', overload='default')>, <OpOverload(op='aten.prod', overload='dim_Dimname')>, <OpOverload(op='aten.hinge_embedding_loss', overload='default')>, <OpOverload(op='aten.std', overload='correction_names')>, <OpOverload(op='aten.mean', overload='names_dim')>, <OpOverload(op='aten.norm', overload='names_ScalarOpt_dim_dtype')>, <OpOverload(op='aten.cumsum', overload='dimname')>, <OpOverload(op='aten.std_mean', overload='names_dim')>, <OpOverload(op='aten.var', overload='names_dim')>, <OpOverload(op='aten.index_add', overload='dimname')>, <OpOverload(op='aten.cat', overload='names')>, <OpOverload(op='aten.std_mean', overload='correction_names')>}

**Comment by justinchuby** [comment]
@Bludator should be fixed by @titaiwangms ' PR. Thank you for reporting the issue!
